src/main/java/org/skriptlang/skript/misc/colours/ExprComplement.java

```diff
@@ -0,0 +1,53 @@
+package org.skriptlang.skript.misc.colours;
+
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Complementary Colours")
+@Description({
+	"Returns the complementary colour of a given colour(s).",
+	"Can optionally use a HSL-based approach if need be, but this is rarely required."
+})
+@Examples({
+	"set {_bluesComplement} to complement of blue",
+	"set {_allComplements} to complementary colour of all colours"
+})
+@Since("INSERT VERSION")
+public class ExprComplement extends SimplePropertyExpression<Color, Color> {
+
+	static {
+		register(ExprComplement.class, Color.class, "[:hsl] complement[ary colo[u]r]", "colors");
```

Comment:
```suggestion
		register(ExprComplement.class, Color.class, "[:hsl] complement[ary] colo[u]r[s]", "colors");
```
Just because complement is a common math and programming term, better to be specific.
Might be worth it to rename the class too